features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt

```diff
@@ -218,4 +222,13 @@ class MessagesPresenter @Inject constructor(
             MessageComposerEvents.SetMode(composerMode)
         )
     }
+
+    @Composable
+    private fun getCanSendEvent(type: MessageEventType): State<Boolean> {
+        val canSendEvent = remember(type) { mutableStateOf(false) }
```

Comment:
I guess we could add this method to matrix-ui like [this](https://github.com/vector-im/element-x-android/blob/6497a3f86ceaa36c9e0c0571bea80e77fcacf221/libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/room/MatrixRoomMembers.kt).

Also you should probably use `produceState` method (which does the same internally)

```
return produceState(initialValue = false, key1 = type){
        room.canSendEvent(type).getOrElse { false }
 }
```

Reply:
Good point, thanks! a8fbcb0b1676d7a447037a68b284c2ea8c9a18e9